### ISSUE TITLE
Memory Leaks

### ISSUE THREAD

**Issue by gruelingpine185** [yeti0904/yterm#11]
This is a log from Leaks, a program that finds memory leaks in a given process. This may be related to #10.

```
Rendered 254 characters
Set option 2004 to h
Set option 12 to l
Set option 2004 to h
Set option 2004 to l
Set option 2004 to h
Process:         yterm [20020]
Path:            /Users/USER/*/yterm
Load Address:    0x10f717000
Identifier:      yterm
Version:         0
Code Type:       X86-64
Platform:        macOS
Parent Process:  leaks [20019]

Date/Time:       2023-08-12 11:55:44.759 -0400
Launch Time:     2023-08-12 11:55:39.951 -0400
OS Version:      macOS 13.4.1 (22F82)
Report Version:  7
Analysis Tool:   /usr/bin/leaks

Physical footprint:         38.5M
Physical footprint (peak):  39.8M
Idle exit:                  untracked
----

leaks Report Version: 4.0, multi-line stacks
Process 20020: 20034 nodes malloced for 4107 KB
Process 20020: 2 leaks for 23568 total leaked bytes.

STACK OF 1 INSTANCE OF 'ROOT LEAK: <malloc in SafeMalloc>':
5   dyld                               0x7ff811dc841f start + 1903
4   yterm                                 0x10f71d80c main + 140  main.c:83
3   yterm                                 0x10f71af0a App_Update + 826  app.c:155
2   yterm                                 0x10f72521a TextScreen_Resize + 42  textScreen.c:154
1   yterm                                 0x10f723b05 SafeMalloc + 21  safe.c:6
0   libsystem_malloc.dylib             0x7ff811f70149 _malloc_zone_malloc_instrumented_or_legacy + 88 
====
    1 (23.0K) ROOT LEAK: <malloc in SafeMalloc 0x7fbeff20a000> [23552]

STACK OF 1 INSTANCE OF 'ROOT LEAK: <malloc in SafeMalloc>':
7   dyld                               0x7ff811dc841f start + 1903
6   yterm                                 0x10f71d7ad main + 45  main.c:79
5   yterm                                 0x10f71a901 App_Init + 193  app.c:28
4   yterm                                 0x10f71aa5a App_AddTab + 186  app.c:61
3   yterm                                 0x10f72432d Terminal_Init + 77  terminal.c:10
2   yterm                                 0x10f725871 DupString + 33  util.c:38
1   yterm                                 0x10f723b05 SafeMalloc + 21  safe.c:6
0   libsystem_malloc.dylib             0x7ff811f70149 _malloc_zone_malloc_instrumented_or_legacy + 88 
====
    1 (16 bytes) ROOT LEAK: <malloc in SafeMalloc 0x7fbefe16d480> [16]  length: 5  "yterm"

...
```

**Comment by MESYETI** [comment]
first leak should be fixed in 5c54634

**Comment by gruelingpine185** [comment]
I'll close the issue. 
For whatever reason the title of the application doesn't get handled. The same issue showed with GLFW and I made an issue for them too. It's been about a year since I made it and it still hasn't been resolved yet. Im convinced that it's a MacOS internals thing.
